pytests/conftest.py

```diff
@@ -94,36 +118,55 @@ def wget(self, url, out, enforce_https=True, fingerprint=None):
         r.raw.decode_content = True
         with open(out, 'wb') as f:
             shutil.copyfileobj(r.raw, f)
-
         return True, None
 
     def run(self, cmd):
-        if cmd[0] is 'tdnf' and self.config['build_dir']:
-            cmd[0] = os.path.join(self.config['build_dir'], 'bin/tdnf')
+        if cmd[0] is 'tdnf':
+            if 'build_dir' in self.config:
+                cmd[0] = os.path.join(self.config['build_dir'], 'bin/tdnf')
+            if cmd[1] is not '--config':
+                cmd.insert(1, '-c')
+                cmd.insert(2, os.path.join(self.config['repo_path'], 'tdnf.conf'))
         use_shell = not isinstance(cmd, list)
         process = subprocess.Popen(cmd, shell=use_shell,
                                    stdout=subprocess.PIPE,
                                    stderr=subprocess.PIPE)
-        process.wait()
+        #process.wait()
         out, err = process.communicate()
-        ret = {}
-        ret['stdout'] = []
-        ret['stdout'] = []
-        ret['retval'] = process.returncode
-        if out:
-            ret['stdout'] = out.decode().split('\n')
-        if err:
-            ret['strerr'] = err.decode().split('\n')
+        stdout = out.decode().strip()
+        stderr = err.decode().strip()
+        retval = process.returncode
+        capture = re.match(r'^Error\((\d+)\) :', stderr)
+        if capture:
+            retval = int(capture.groups()[0])
 
+        ret = {}
+        if stdout:
+            ret['stdout'] = stdout.split('\n')
+        else:
+            ret['stdout'] = []
+        if stderr:
+            ret['stderr'] = stderr.split('\n')
+        else:
+            ret['stderr'] = []
+        ret['retval'] = retval
         return ret
 
-def backup_config_files():
-    # Backup /etc/yum.repos.d/* and /etc/tdnf/tdnf.conf
-    pass
-
-def restore_config_files():
-    # Restore /etc/yum.repos.d/* and /etc/tdnf/tdnf.conf
-    pass
+def backup_files(conf_dir):
+    for file in os.listdir(conf_dir):
+        if file.endswith('.bak'):
+            continue
+        src  = os.path.join(conf_dir, file)
+        dest = os.path.join(conf_dir, os.path.basename(file) + '.bak')
+        shutil.copyfile(src, dest)
+
+def restore_files(conf_dir):
```

Comment:
I do not find this function's usage in here. Is this used somewhere else?

Reply:
It was an unused method. Removed.